src/librustc_mir/build/matches/test.rs

```diff
@@ -162,12 +162,19 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             }
 
             TestKind::SwitchInt { switch_ty, ref options, indices: _ } => {
-                let otherwise = self.cfg.start_new_block();
-                let targets: Vec<_> =
-                    options.iter()
-                           .map(|_| self.cfg.start_new_block())
-                           .chain(Some(otherwise))
-                           .collect();
+                let targets: Vec<_> = match switch_ty.sty {
+                    ty::TyBool if options.len() == 2
```

Comment:
seems...ok to me :) might be nice to be a touch more general (e.g., if there is an exhaustive `u8`)